presto-main/src/test/java/com/facebook/presto/server/TestResourceGroupStateInfo.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.resourceGroups.ResourceGroupId;
+import com.google.common.collect.ImmutableList;
+import io.airlift.json.JsonCodec;
+import io.airlift.units.DataSize;
+import org.joda.time.DateTime;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+import java.util.OptionalDouble;
+
+import static com.facebook.presto.execution.QueryState.RUNNING;
+import static com.facebook.presto.spi.resourceGroups.ResourceGroupState.CAN_RUN;
+import static io.airlift.units.DataSize.Unit.BYTE;
+import static org.testng.Assert.assertEquals;
+
+public class TestResourceGroupStateInfo
+{
+    @Test
+    public void testResourceGroupStateInfoJson()
```

Comment:
Could you add a test to the existing classes that validates data derived from actual decisions made by the resource group manager? Similar to the stuff you did with waitingQueuedQueries?